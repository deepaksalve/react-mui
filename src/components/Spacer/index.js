import React from "react";
import Box from "@mui/material/Box";

export default function Spacer({ dx, dy }) {
  const sx = {};

  if (dx) {
    sx.paddingTop = dx;
    sx.paddingBottom = dx;
  }

  if (dy) {
    sx.paddingLeft = dy;
    sx.paddingRight = dy;
  }

  return <Box sx={sx} data-testid="spacer" />;
}
