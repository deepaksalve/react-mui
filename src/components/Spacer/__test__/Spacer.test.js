import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Spacer from "../";

describe("Spacer", () => {
  test("renders", () => {
    render(<Spacer />);
    const elm = screen.getByTestId("spacer");

    expect(elm).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const tree = renderer.create(<Spacer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
