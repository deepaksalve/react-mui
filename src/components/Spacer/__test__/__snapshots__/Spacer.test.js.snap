// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer matches snapshot 1`] = `
<div
  className="MuiBox-root css-0"
  data-testid="spacer"
/>
`;
